### PR TITLE
feat(formula): align to template formula

### PR DESCRIPTION
**Pull request by noelmcloughlin**
### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [x] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [x] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [x] `[docs]`     Documentation changes
- [x] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

YES

### Describe the changes you're proposing

Using IntelliJ-formula as a template, align to template-formula.

### Documentation checklist

- [x] Updated the `README` (e.g. `Available states`).
- [x] Updated `pillar.example`.

### Testing checklist

- [x] Included in Kitchen (i.e. under `state_top`).
- [x] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [x] Updated the relevant test pillar.

### CODE REVIEW

**Comment by noelmcloughlin** [comment]
Trace on MacOS
```
  ID: pycharm-macos-app-clean-files
    Function: file.absent
        Name: /tmp/jetbrains/pycharm
      Result: True
     Comment: File /tmp/jetbrains/pycharm is not present
     Started: 14:49:26.990998
    Duration: 0.512 ms
     Changes:   
----------
          ID: pycharm-macos-app-clean-files
    Function: file.absent
        Name: /Applications/Pycharm.app
      Result: True
     Comment: File /Applications/Pycharm.app is not present
     Started: 14:49:26.991644
    Duration: 0.342 ms
     Changes:   
----------
          ID: pycharm-config-clean-file-absent
    Function: file.absent
        Name: /tmp/dummy_list_item
      Result: True
     Comment: File /tmp/dummy_list_item is not present
     Started: 14:49:26.992197
    Duration: 0.352 ms
     Changes:   
----------
          ID: pycharm-config-clean-file-absent
    Function: file.absent
        Name: /etc/defaults/pycharm.sh
      Result: True
     Comment: File /etc/defaults/pycharm.sh is not present
     Started: 14:49:26.992799
    Duration: 0.942 ms
     Changes:   
----------
          ID: pycharm-config-clean-file-absent
    Function: file.absent
        Name: /Users/nmcloughlin/Desktop/Pycharm CE
      Result: True
     Comment: Removed file /Users/nmcloughlin/Desktop/Pycharm CE
     Started: 14:49:26.993984
    Duration: 12.068 ms
     Changes:   
              ----------
              removed:
                  /Users/nmcloughlin/Desktop/Pycharm CE
----------
          ID: pycharm-macos-app-install-curl
    Function: file.directory
        Name: /tmp/jetbrains/pycharm
      Result: True
     Comment: Directory /tmp/jetbrains/pycharm updated
     Started: 14:49:27.006220
    Duration: 2.269 ms
     Changes:   
              ----------
              /tmp/jetbrains/pycharm:
                  New Dir
----------
          ID: pycharm-macos-app-install-curl
    Function: pkg.installed
        Name: curl
      Result: True
     Comment: All specified packages are already installed
     Started: 14:49:31.626974
    Duration: 11248.389 ms
     Changes:   
----------
          ID: pycharm-macos-app-install-curl
    Function: cmd.run
        Name: curl -Lo /tmp/jetbrains/pycharm/pycharm-2020.1 https://download.jetbrains.com/python/pycharm-community-2020.1.dmg
      Result: True
     Comment: Command "curl -Lo /tmp/jetbrains/pycharm/pycharm-2020.1 https://download.jetbrains.com/python/pycharm-community-2020.1.dmg" run
     Started: 14:49:42.879344
    Duration: 666600.319 ms
     Changes:   
              ----------
              pid:
                  5338
              retcode:
                  0
              stderr:
                    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                                   Dload  Upload   Total   Spent    Left  Speed
                  
                    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
                    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
                  100   138  100   138    0     0    309      0 --:--:-- --:--:-- --:--:--   308
                  
                    0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
                    0  360M    0  117k    0     0  56392      0  1:51:37  0:00:02  1:51:35  117k
                    0  360M    0  660k    0     0   193k      0  0:31:46  0:00:03  0:31:43  289k
     
       << CUT >>

                   99  360M   99  359M    0     0   552k      0  0:11:07  0:11:05  0:00:02  672k
                   99  360M   99  359M    0     0   552k      0  0:11:07  0:11:06  0:00:01  635k
                  100  360M  100  360M    0     0   553k      0  0:11:06  0:11:06 --:--:--  741k
              stdout:
----------
          ID: pycharm-macos-app-install-checksum
    Function: module.run
        Name: file.check_hash
      Result: True
     Comment: onlyif condition is false
     Started: 15:00:49.485185
    Duration: 909.815 ms
     Changes:   
----------
          ID: pycharm-macos-app-install-checksum
    Function: file.absent
        Name: /tmp/jetbrains/pycharm/pycharm-2020.1
      Result: True
     Comment: State was not run because onfail req did not change
     Started: 15:00:50.396295
    Duration: 0.007 ms
     Changes:   
----------
          ID: pycharm-macos-app-install-macpackage
    Function: macpackage.installed
        Name: /tmp/jetbrains/pycharm/pycharm-2020.1
      Result: True
     Comment: PyCharm CE.app installed
     Started: 15:00:50.396641
    Duration: 25264.116 ms
     Changes:   
              ----------
              installed:
                  - PyCharm CE.app
----------
          ID: pycharm-macos-app-install-macpackage
    Function: file.managed
        Name: /tmp/mac_shortcut.sh
      Result: True
     Comment: File /tmp/mac_shortcut.sh updated
     Started: 15:01:15.661058
    Duration: 42.897 ms
     Changes:   
              ----------
              diff:
                  --- 
                  +++ 
                  @@ -1,5 +1,5 @@
                   #!/usr/bin/env bash
                   
                  -Source="/Applications/Intellij IDEA CE.app"
                  +Source="/Applications/Pycharm CE.app"
                   Destination="/Users/nmcloughlin/Desktop"
                   /usr/bin/osascript -e "tell application \"Finder\" to make alias file to POSIX file \"$Source\" at POSIX file \"$Destination\""
----------
          ID: pycharm-macos-app-install-macpackage
    Function: cmd.run
        Name: /tmp/mac_shortcut.sh
      Result: True
     Comment: Command "/tmp/mac_shortcut.sh" run
     Started: 15:01:15.704580
    Duration: 1397.742 ms
     Changes:   
              ----------
              pid:
                  7577
              retcode:
                  0
              stderr:
              stdout:
                  alias file PyCharm CE of folder Desktop of folder nmcloughlin of folder Users of startup disk
----------
          ID: pycharm-config-file-file-managed-environ_file
    Function: file.managed
        Name: /etc/defaults/pycharm.sh
      Result: True
     Comment: File /etc/defaults/pycharm.sh updated
     Started: 15:01:17.102864
    Duration: 1660.913 ms
     Changes:   
              ----------
              diff:
                  New file
              group:
                  nmcloughlin
              mode:
                  0644
              user:
                  nmcloughlin

Summary for local
-------------
Succeeded: 14 (changed=7)
Failed:     0
-------------
Total states run:     14
```

**Comment by myii** [comment]
@noelmcloughlin I've finished checking this with the `ssf-formula`:

* https://github.com/myii/ssf-formula/pull/180

I've run it in my own fork with the standardised changes, including updating to the latest images -- all is working well:

* https://travis-ci.com/github/myii/jetbrains-pycharm-formula/builds/164853358

Here's the actual commit to be added here:

* https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6
  - I've got some feedback to share about this, I still need to decide whether to add it here or as comments on that commit.

I wasn't able to add it here because of permissions (deselected maintainer edits?).

There's also the merge conflicts to be resolved here due to #7; it would be helpful if PRs were commenced from the latest commit in `master` (or at least rebased afterwards) to avoid conflicts and ease reviewing/amendments.

File: .travis.yml
Comment (myii) [inline .travis.yml:50]: Nothing being done with `shellcheck` here, will be resolved when commit is added here:

https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6#diff-354f30a63fb0907d4ad57269548329e3R51-R52

File: .yamllint
Comment (myii) [inline .yamllint]: Unnecessary, will be resolved when commit is added here:

https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6#diff-b97c55dd898003930e2858fadd3b5c9cL16-L17

File: FORMULA
Comment (myii) [inline FORMULA]: We're no longer supporting < `2019.2`, will be resolved when commit is added here:

https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6#diff-adc4995cd52208dde39433fb069ddd61R6

File: Gemfile
Comment (myii) [inline Gemfile]: No longer necessary since fixed upstream and we're also adding the `Gemfile.lock` now, will be resolved when commit is added here:

https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6#diff-8b7db4d5cc4b8f6dc8feb7030baa2478L8-L11

File: docs/README.rst
Comment (myii) [inline docs/README.rst]: Headings markers should match heading text length, will be resolved when commit is added here:

https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6#diff-8ab5636c6c18641e5c58358ef489e0f7R4

File: docs/README.rst
Comment (myii) [inline docs/README.rst]: Using `depth` after your PR to the `template-formula`(!), will be resolved when commit is added here:

https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6#diff-8ab5636c6c18641e5c58358ef489e0f7R20

File: docs/TOFS_pattern.rst
Comment (myii) [inline docs/TOFS_pattern.rst]: We use the same `docs/TOFS_pattern.rst` in all formulas, will be resolved when commit is added here:

https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6#diff-bf838b5bad3aeeb4d43d3e986622d832R263-R273

File: kitchen.yml
Comment (myii) [inline kitchen.yml:161]: The general standard we decided upon for test pillars is:

* `test/salt/pillar/<suite-name-or-similar>.sls`
  - We've not been too picky about the name of the actual file, as long as it's clear
  - Personally, I believe it should always contain the InSpec suite name at the very least, if that's not already sufficient

You can see how this is being used across the org by searching for `pillars_from_files` in the `formulas.yaml` of the `ssf-formula`:

* https://github.com/myii/ssf-formula/blob/master/ssf/formulas.yaml

In this PR, that would also affect the line further below.

File: kitchen.yml
Comment (myii) [inline kitchen.yml]: Adding inclusions and exclusions (particularly inclusions) when there are multiple suites is important, otherwise it can lead to massive lists when running `kitchen list` -- and most of these won't even work due to the nature of the suites; will be resolved when commit is added here:

https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6#diff-fdbca9ed07a634829a3548f874dc925aR146-R168

File: test/integration/arch/README.md
Comment (myii) [inline test/integration/arch/README.md]: The actual suite name needs to be used throughout this file, will be resolved when commit is added here:

https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6#diff-19a94498bc41afb14c871deec9e4b473L1-R43

File: test/integration/arch/inspec.yml
Comment (myii) [inline test/integration/arch/inspec.yml]: The name and summary should be appropriate for each suite, will be resolved when commit is added here:

https://github.com/myii/jetbrains-pycharm-formula/commit/a3bac46edc0678d4436b425e3929d5c04aea72f6#diff-e12e128804f69778e0d42bd5d468d7a8L4-R9

**Review by myii — CHANGES_REQUESTED** [review]
Here are the inline comments I mentioned earlier.  The only actual change that's needed from your side is the paths to the test pillar files (and the respective updates to `kitchen.yml`).  The rest will be automatically added from the commit that I've linked to (although, I'd have to update that commit if you do make the change to the location of the test pillar files).

I've gone to a significant effort to give this feedback here, in order to ease the process of your `semantic-release` PRs being merged (and reviewed!).

File: .travis.yml
Comment (noelmcloughlin) [reply .travis.yml:50]: Hi, @myii Thanks for the effort in reviewing and providing the feedback.

**Comment by noelmcloughlin** [comment]
Hi @myii 
I've gone to considerable effort to align all my new formulas (new PRs) and updated formulas (existing PRs) with all your comments.  The only one not actioned is changing `test/salt/<suite-name-or-similar>/pillar.sls` to `test/salt/pillar/<suite-name-or-similar>.sls` because I missed that comment.   That is something I can look at in future I guess. I fixed the conflicts.

This work should make reviews easier because the comments here are actioned in all my open PRs.

**Comment by myii** [comment]
> I've gone to considerable effort to align all my new formulas (new PRs) and updated formulas (existing PRs) with all your comments. The only one not actioned is changing `test/salt/<suite-name-or-similar>/pillar.sls` to `test/salt/pillar/<suite-name-or-similar>.sls` because I missed that comment. That is something I can look at in future I guess. I fixed the conflicts.
> 
> This work should make reviews easier because the comments here are actioned in all my open PRs.

Thanks for your efforts, @noelmcloughlin.  I'll keep working my way through the PRs.

In the meantime, I've updated the commit to be added here:

* https://github.com/myii/jetbrains-pycharm-formula/commit/d2c58246ec5b07e0dd0b8038d8882854162ce00e
* https://travis-ci.com/github/myii/jetbrains-pycharm-formula/builds/165078462

I still couldn't add it here because of permissions.  So please could you add that commit here yourself, or make sure `Allow edits by maintainers` is selected and then let me know so that I can push it here.

**Comment by noelmcloughlin** [comment]
Can you try now? I updated access  Allow edits by maintainers seems to be
ON now.

**Comment by myii** [comment]
@noelmcloughlin Still no cigar:

```bash
ERROR: Permission to noelmcloughlin/jetbrains-pycharm-formula.git denied to myii.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Not sure why it's being problematic, I've been using it fine elsewhere, as you know.  If you add my fork locally, it's pretty simple to cherry-pick my commit and push it here.

**Review by myii — APPROVED** [review]

**Comment by myii** [comment]
Thanks, @noelmcloughlin -- we finally got there.

Note, that `centos-8` job `Ran for 10 min 7 sec ` -- may need to keep an eye on that.  I prefer to set `retry: 1` for our Travis jobs (in the test pillar), to prevent them holding up other instances across the org, since we've only got 5 concurrent in total.

**Pull request merged**
